src/main/java/com/avail/interpreter/levelTwo/operation/L2_CREATE_CONTINUATION.java

```diff
@@ -156,47 +137,33 @@ public void translateToJVM (
 		final MethodVisitor method,
 		final L2Instruction instruction)
 	{
+		assert this == instruction.operation();
 		final L2ReadBoxedOperand function = instruction.operand(0);
 		final L2ReadBoxedOperand caller = instruction.operand(1);
 		final L2IntImmediateOperand levelOnePC = instruction.operand(2);
 		final L2IntImmediateOperand levelOneStackp = instruction.operand(3);
 		final L2ReadBoxedVectorOperand slots = instruction.operand(4);
 		final L2WriteBoxedOperand destReg = instruction.operand(5);
-		final L2PcOperand onRamp = instruction.operand(6);
-		final L2PcOperand fallThrough = instruction.operand(7);
+		final L2ReadIntOperand labelIntReg = instruction.operand(6);
+		final L2ReadBoxedOperand registerDumpReg = instruction.operand(7);
 
 		// :: continuation = createContinuationExceptFrame(
 		// ::    function,
 		// ::    caller,
+		// ::    registerDump
 		// ::    levelOnePC,
 		// ::    levelOneStackp,
 		// ::    interpreter.chunk,
 		// ::    onRampOffset);
 		translator.load(method, function.register());
 		translator.load(method, caller.register());
+		translator.load(method, registerDumpReg.register());
 		translator.literal(method, levelOnePC.value);
 		translator.literal(method, levelOneStackp.value);
 		translator.loadInterpreter(method);
-		method.visitFieldInsn(
-			GETFIELD,
-			getInternalName(Interpreter.class),
-			"chunk",
-			getDescriptor(L2Chunk.class));
-		translator.intConstant(method, onRamp.offset());
-		method.visitMethodInsn(
-			INVOKESTATIC,
-			getInternalName(ContinuationDescriptor.class),
-			"createContinuationExceptFrame",
-			getMethodDescriptor(
-				getType(A_Continuation.class),
-				getType(A_Function.class),
-				getType(A_Continuation.class),
-				INT_TYPE,
-				INT_TYPE,
-				getType(L2Chunk.class),
-				INT_TYPE),
-			false);
-		method.visitTypeInsn(CHECKCAST, getInternalName(AvailObject.class));
+		chunkField.generateRead(method);
+		translator.load(method, labelIntReg.register());
+		createContinuationExceptFrameMethod.generateCall(method);
```

Comment:
Wow, so much cleaner!

Reply:
Thanks, I agree.  And safer, since the method signature and instance/static -ness are checked when the containing class is loaded, rather than when the verifier gags cryptically.